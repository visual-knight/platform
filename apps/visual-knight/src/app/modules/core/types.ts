import gql from 'graphql-tag';
import { Injectable } from '@angular/core';
import * as Apollo from 'apollo-angular';
export type Maybe<T> = T | null;
/** All built-in and custom scalars, mapped to their actual values */
export type Scalars = {
  ID: string;
  String: string;
  Boolean: boolean;
  Int: number;
  Float: number;
  /** Use JavaScript Date object for date/time fields. */
  DateTime: any;
  /** The `JSON` scalar type represents JSON values as specified by [ECMA-404](http://www.ecma-international.org/publications/files/ECMA-ST/ECMA-404.pdf). */
  JSON: any;
};

export type AuthPayload = {
  __typename?: 'AuthPayload';
  token: AuthToken;
  user: User;
};

export type AuthToken = {
  __typename?: 'AuthToken';
  expiresIn: Scalars['Int'];
  accessToken: Scalars['String'];
};

export type InvokeTestsession = {
  __typename?: 'InvokeTestsession';
  url: Scalars['String'];
  testSessionId: Scalars['String'];
};

export type Mutation = {
  __typename?: 'Mutation';
  login: AuthPayload;
  signup: AuthPayload;
  verifyEmail: Scalars['Boolean'];
  resendVerifyEmail: Scalars['Boolean'];
  changePassword: Scalars['Boolean'];
  forgotPassword: Scalars['Boolean'];
  resetPassword: Scalars['Boolean'];
  deleteUser: User;
  updateUser: User;
  inviteNewUser: User;
  completeInvitation: AuthPayload;
  createProject: ProjectType;
  deleteProject: ProjectType;
  updateProject: ProjectType;
  deleteTestSession: TestSessionType;
  updateTestSession: TestSessionType;
  deleteTest: TestType;
  deleteVariation: VariationType;
  invokeTestSession: InvokeTestsession;
};

export type MutationLoginArgs = {
  password: Scalars['String'];
  email: Scalars['String'];
};

export type MutationSignupArgs = {
  password: Scalars['String'];
  email: Scalars['String'];
};

export type MutationVerifyEmailArgs = {
  token: Scalars['String'];
};

export type MutationChangePasswordArgs = {
  password: Scalars['String'];
};

export type MutationForgotPasswordArgs = {
  email: Scalars['String'];
};

export type MutationResetPasswordArgs = {
  token: Scalars['String'];
  password: Scalars['String'];
};

export type MutationDeleteUserArgs = {
  id: Scalars['String'];
};

export type MutationUpdateUserArgs = {
  data: UpdateUserInput;
};

export type MutationInviteNewUserArgs = {
  email: Scalars['String'];
};

export type MutationCompleteInvitationArgs = {
  password: Scalars['String'];
  token: Scalars['String'];
};

export type MutationCreateProjectArgs = {
  data: ProjectDataArgs;
};

export type MutationDeleteProjectArgs = {
  projectId: Scalars['String'];
};

export type MutationUpdateProjectArgs = {
  data: ProjectDataArgs;
  projectId: Scalars['String'];
};

export type MutationDeleteTestSessionArgs = {
  testSessionId: Scalars['String'];
};

export type MutationUpdateTestSessionArgs = {
  data: TestSessionDataArgs;
  testSessionId: Scalars['String'];
};

export type MutationDeleteTestArgs = {
  testId: Scalars['String'];
};

export type MutationDeleteVariationArgs = {
  variationId: Scalars['String'];
};

export type MutationInvokeTestSessionArgs = {
  autoBaseline: Scalars['Boolean'];
  capabilities: Scalars['JSON'];
  misMatchTolerance: Scalars['Float'];
  project: Scalars['String'];
  testname: Scalars['String'];
};

export type ProjectDataArgs = {
  name?: Maybe<Scalars['String']>;
  description?: Maybe<Scalars['String']>;
};

export type ProjectType = {
  __typename?: 'ProjectType';
  id: Scalars['ID'];
  name: Scalars['String'];
  description?: Maybe<Scalars['String']>;
};

export type Query = {
  __typename?: 'Query';
  me: User;
  project: ProjectType;
  projects: Array<ProjectType>;
  projectsCount: Scalars['Int'];
  testSession: TestSessionType;
  testSessions: Array<TestSessionType>;
  testSessionsCount: Scalars['Int'];
  test: TestType;
  tests: Array<TestType>;
  testsCount: Scalars['Int'];
  variation: VariationType;
  variations: Array<VariationType>;
  variationsCount: Scalars['Int'];
  testSessionWatch?: Maybe<TestSessionComparison>;
};

export type QueryProjectArgs = {
  projectId: Scalars['String'];
};

export type QueryTestSessionArgs = {
  testSessionId: Scalars['String'];
};

export type QueryTestSessionsArgs = {
  where?: Maybe<TestSessionDataArgs>;
};

export type QueryTestSessionsCountArgs = {
  where?: Maybe<TestSessionDataArgs>;
};

export type QueryTestArgs = {
  testId: Scalars['String'];
};

export type QueryVariationArgs = {
  variationId: Scalars['String'];
};

export type QueryVariationsArgs = {
  testId: Scalars['String'];
};

export type QueryVariationsCountArgs = {
  testId: Scalars['String'];
};

export type QueryTestSessionWatchArgs = {
  testSessionId: Scalars['String'];
};

export type TestSessionComparison = {
  __typename?: 'TestSessionComparison';
  id: Scalars['ID'];
  diffImageKey?: Maybe<Scalars['String']>;
  imageKey?: Maybe<Scalars['String']>;
  misMatchPercentage?: Maybe<Scalars['Float']>;
  misMatchTolerance: Scalars['Float'];
  isSameDimensions?: Maybe<Scalars['Boolean']>;
  state: Scalars['String'];
  stateComment?: Maybe<Scalars['String']>;
  autoBaseline: Scalars['Boolean'];
  createdAt: Scalars['DateTime'];
  link: Scalars['String'];
};

export type TestSessionDataArgs = {
  id?: Maybe<Scalars['ID']>;
  diffImageKey?: Maybe<Scalars['String']>;
  imageKey?: Maybe<Scalars['String']>;
  misMatchPercentage?: Maybe<Scalars['Float']>;
  misMatchTolerance: Scalars['Float'];
  isSameDimensions?: Maybe<Scalars['Boolean']>;
  state: Scalars['String'];
  stateComment?: Maybe<Scalars['String']>;
  autoBaseline: Scalars['Boolean'];
};

export type TestSessionType = {
  __typename?: 'TestSessionType';
  id: Scalars['ID'];
  diffImageKey?: Maybe<Scalars['String']>;
  imageKey?: Maybe<Scalars['String']>;
  misMatchPercentage?: Maybe<Scalars['Float']>;
  misMatchTolerance: Scalars['Float'];
  isSameDimensions?: Maybe<Scalars['Boolean']>;
  state: Scalars['String'];
  stateComment?: Maybe<Scalars['String']>;
  autoBaseline: Scalars['Boolean'];
  createdAt: Scalars['DateTime'];
};

export type TestType = {
  __typename?: 'TestType';
  id: Scalars['ID'];
  name: Scalars['String'];
  project?: Maybe<ProjectType>;
  variations: Array<VariationType>;
};

export type UpdateUserInput = {
  email: Scalars['String'];
  forename?: Maybe<Scalars['String']>;
  lastname?: Maybe<Scalars['String']>;
};

export type User = {
  __typename?: 'User';
  id: Scalars['ID'];
  email: Scalars['String'];
  forename?: Maybe<Scalars['String']>;
  lastname?: Maybe<Scalars['String']>;
  apiKey: Scalars['String'];
  active: Scalars['Boolean'];
};

export type VariationType = {
  __typename?: 'VariationType';
  id: Scalars['ID'];
  browserName: Scalars['String'];
  deviceName: Scalars['String'];
  createdAt: Scalars['DateTime'];
  additionalData?: Maybe<Scalars['JSON']>;
  baseline?: Maybe<TestSessionType>;
  testSessions: Array<TestSessionType>;
};
export type CurrentUserQueryVariables = {};

export type CurrentUserQuery = { __typename?: 'Query' } & {
  me: { __typename?: 'User' } & UserDataFragment;
};

export type SignupMutationVariables = {
  email: Scalars['String'];
  password: Scalars['String'];
};

export type SignupMutation = { __typename?: 'Mutation' } & {
  signup: { __typename?: 'AuthPayload' } & {
    token: { __typename?: 'AuthToken' } & AuthTokenFragment;
    user: { __typename?: 'User' } & UserDataFragment;
  };
};

export type LoginMutationVariables = {
  email: Scalars['String'];
  password: Scalars['String'];
};

export type LoginMutation = { __typename?: 'Mutation' } & {
  login: { __typename?: 'AuthPayload' } & {
    token: { __typename?: 'AuthToken' } & AuthTokenFragment;
    user: { __typename?: 'User' } & UserDataFragment;
  };
};

export type ForgotPasswordMutationVariables = {
  email: Scalars['String'];
};

export type ForgotPasswordMutation = { __typename?: 'Mutation' } & Pick<
  Mutation,
  'forgotPassword'
>;

export type ResetPasswordMutationVariables = {
  password: Scalars['String'];
  token: Scalars['String'];
};

export type ResetPasswordMutation = { __typename?: 'Mutation' } & Pick<
  Mutation,
  'resetPassword'
>;

export type AuthTokenFragment = { __typename?: 'AuthToken' } & Pick<
  AuthToken,
  'expiresIn' | 'accessToken'
>;

export type UserDataFragment = { __typename?: 'User' } & Pick<
  User,
  'id' | 'active' | 'email' | 'apiKey'
>;

export type AllProjectsQueryVariables = {};

export type AllProjectsQuery = { __typename?: 'Query' } & {
  projects: Array<{ __typename?: 'ProjectType' } & ProjectDataFragment>;
};

export type AddProjectMutationVariables = {
  name: Scalars['String'];
  description?: Maybe<Scalars['String']>;
};

export type AddProjectMutation = { __typename?: 'Mutation' } & {
  createProject: { __typename?: 'ProjectType' } & ProjectDataFragment;
};

export type DeleteProjectMutationVariables = {
  projectId: Scalars['String'];
};

export type DeleteProjectMutation = { __typename?: 'Mutation' } & {
  deleteProject: { __typename?: 'ProjectType' } & Pick<ProjectType, 'id'>;
};

export type ProjectDataFragment = { __typename?: 'ProjectType' } & Pick<
  ProjectType,
  'id' | 'name' | 'description'
>;

export type AllTestsQueryVariables = {};

export type AllTestsQuery = { __typename?: 'Query' } & {
  tests: Array<{ __typename?: 'TestType' } & TestDataFragment>;
};

export type DeleteTestMutationVariables = {
  id: Scalars['String'];
};

export type DeleteTestMutation = { __typename?: 'Mutation' } & {
  deleteTest: { __typename?: 'TestType' } & Pick<TestType, 'id'>;
};

export type TestDataFragment = { __typename?: 'TestType' } & Pick<
  TestType,
  'id' | 'name'
> & {
    project: Maybe<{ __typename?: 'ProjectType' } & Pick<ProjectType, 'id'>>;
    variations: Array<
      { __typename?: 'VariationType' } & Pick<
        VariationType,
        'id' | 'createdAt' | 'browserName' | 'deviceName'
      > & {
          testSessions: Array<
            { __typename?: 'TestSessionType' } & Pick<
              TestSessionType,
              'id' | 'state'
            >
          >;
        }
    >;
  };

export type GetVariationQueryVariables = {
  variationId: Scalars['String'];
};

export type GetVariationQuery = { __typename?: 'Query' } & {
  variation: { __typename?: 'VariationType' } & VariationDataFragment;
};

export type AllVariationsQueryVariables = {
  testId: Scalars['String'];
};

export type AllVariationsQuery = { __typename?: 'Query' } & {
  variations: Array<{ __typename?: 'VariationType' } & VariationDataFragment>;
};

export type DeleteVariationMutationVariables = {
  id: Scalars['String'];
};

export type DeleteVariationMutation = { __typename?: 'Mutation' } & {
  deleteVariation: { __typename?: 'VariationType' } & Pick<VariationType, 'id'>;
};

export type VariationDataFragment = { __typename?: 'VariationType' } & Pick<
  VariationType,
  'id' | 'deviceName' | 'additionalData' | 'browserName'
> & {
    testSessions: Array<
      { __typename?: 'TestSessionType' } & Pick<
        TestSessionType,
        | 'id'
        | 'diffImageKey'
        | 'imageKey'
        | 'misMatchPercentage'
        | 'misMatchTolerance'
        | 'createdAt'
        | 'state'
        | 'stateComment'
        | 'autoBaseline'
      >
    >;
  };
export const AuthTokenFragmentDoc = gql`
  fragment AuthToken on AuthToken {
    expiresIn
    accessToken
  }
`;
export const UserDataFragmentDoc = gql`
  fragment UserData on User {
    id
    active
    email
    apiKey
  }
`;
export const ProjectDataFragmentDoc = gql`
  fragment ProjectData on ProjectType {
    id
    name
    description
  }
`;
export const TestDataFragmentDoc = gql`
  fragment TestData on TestType {
    id
    name
    project {
      id
    }
    variations {
      id
      createdAt
      browserName
      deviceName
      testSessions {
        id
        state
      }
    }
  }
`;
export const VariationDataFragmentDoc = gql`
  fragment VariationData on VariationType {
    id
    deviceName
    additionalData
    browserName
    testSessions {
      id
      diffImageKey
      imageKey
      misMatchPercentage
      misMatchTolerance
      createdAt
      state
      stateComment
      autoBaseline
    }
  }
`;
export const CurrentUserDocument = gql`
  query currentUser {
    me {
      ...UserData
    }
  }
  ${UserDataFragmentDoc}
`;

@Injectable({
  providedIn: 'root'
})
export class CurrentUserGQL extends Apollo.Query<
  CurrentUserQuery,
  CurrentUserQueryVariables
> {
  document = CurrentUserDocument;
}
export const SignupDocument = gql`
  mutation signup($email: String!, $password: String!) {
    signup(email: $email, password: $password) {
      token {
        ...AuthToken
      }
      user {
        ...UserData
      }
    }
  }
  ${AuthTokenFragmentDoc}
  ${UserDataFragmentDoc}
`;

@Injectable({
  providedIn: 'root'
})
export class SignupGQL extends Apollo.Mutation<
  SignupMutation,
  SignupMutationVariables
> {
  document = SignupDocument;
}
export const LoginDocument = gql`
  mutation login($email: String!, $password: String!) {
    login(email: $email, password: $password) {
      token {
        ...AuthToken
      }
      user {
        ...UserData
      }
    }
  }
  ${AuthTokenFragmentDoc}
  ${UserDataFragmentDoc}
`;

@Injectable({
  providedIn: 'root'
})
export class LoginGQL extends Apollo.Mutation<
  LoginMutation,
  LoginMutationVariables
> {
  document = LoginDocument;
}
export const ForgotPasswordDocument = gql`
  mutation forgotPassword($email: String!) {
    forgotPassword(email: $email)
  }
`;

@Injectable({
  providedIn: 'root'
})
export class ForgotPasswordGQL extends Apollo.Mutation<
  ForgotPasswordMutation,
  ForgotPasswordMutationVariables
> {
  document = ForgotPasswordDocument;
}
export const ResetPasswordDocument = gql`
  mutation resetPassword($password: String!, $token: String!) {
    resetPassword(token: $token, password: $password)
  }
`;

@Injectable({
  providedIn: 'root'
})
export class ResetPasswordGQL extends Apollo.Mutation<
  ResetPasswordMutation,
  ResetPasswordMutationVariables
> {
  document = ResetPasswordDocument;
}
export const AllProjectsDocument = gql`
  query allProjects {
    projects {
      ...ProjectData
    }
  }
  ${ProjectDataFragmentDoc}
`;

@Injectable({
  providedIn: 'root'
})
export class AllProjectsGQL extends Apollo.Query<
  AllProjectsQuery,
  AllProjectsQueryVariables
> {
  document = AllProjectsDocument;
}
export const AddProjectDocument = gql`
  mutation addProject($name: String!, $description: String) {
    createProject(data: { name: $name, description: $description }) {
      ...ProjectData
    }
  }
  ${ProjectDataFragmentDoc}
`;

@Injectable({
  providedIn: 'root'
})
export class AddProjectGQL extends Apollo.Mutation<
  AddProjectMutation,
  AddProjectMutationVariables
> {
  document = AddProjectDocument;
}
export const DeleteProjectDocument = gql`
  mutation deleteProject($projectId: String!) {
    deleteProject(projectId: $projectId) {
      id
    }
  }
`;

@Injectable({
  providedIn: 'root'
})
export class DeleteProjectGQL extends Apollo.Mutation<
  DeleteProjectMutation,
  DeleteProjectMutationVariables
> {
  document = DeleteProjectDocument;
}
export const AllTestsDocument = gql`
  query allTests {
    tests {
      ...TestData
    }
  }
  ${TestDataFragmentDoc}
`;

@Injectable({
  providedIn: 'root'
})
export class AllTestsGQL extends Apollo.Query<
  AllTestsQuery,
  AllTestsQueryVariables
> {
  document = AllTestsDocument;
}
export const DeleteTestDocument = gql`
  mutation deleteTest($id: String!) {
    deleteTest(testId: $id) {
      id
    }
  }
`;

@Injectable({
  providedIn: 'root'
})
export class DeleteTestGQL extends Apollo.Mutation<
  DeleteTestMutation,
  DeleteTestMutationVariables
> {
  document = DeleteTestDocument;
}
export const GetVariationDocument = gql`
  query getVariation($variationId: String!) {
    variation(variationId: $variationId) {
      ...VariationData
    }
  }
  ${VariationDataFragmentDoc}
`;

@Injectable({
  providedIn: 'root'
})
export class GetVariationGQL extends Apollo.Query<
  GetVariationQuery,
  GetVariationQueryVariables
> {
  document = GetVariationDocument;
}
export const AllVariationsDocument = gql`
  query allVariations($testId: String!) {
    variations(testId: $testId) {
      ...VariationData
    }
  }
  ${VariationDataFragmentDoc}
`;

@Injectable({
  providedIn: 'root'
})
export class AllVariationsGQL extends Apollo.Query<
  AllVariationsQuery,
  AllVariationsQueryVariables
> {
  document = AllVariationsDocument;
}
export const DeleteVariationDocument = gql`
  mutation deleteVariation($id: String!) {
    deleteVariation(variationId: $id) {
      id
    }
  }
`;

@Injectable({
  providedIn: 'root'
})
export class DeleteVariationGQL extends Apollo.Mutation<
  DeleteVariationMutation,
  DeleteVariationMutationVariables
> {
  document = DeleteVariationDocument;
}
