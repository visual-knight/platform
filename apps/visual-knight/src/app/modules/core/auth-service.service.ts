import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import {
  User,
  CurrentUserGQL,
  SignupGQL,
  LoginGQL,
  ForgotPasswordGQL,
  ResetPasswordGQL
} from './types';
import { map, tap } from 'rxjs/operators';
import { GraphQLError } from 'graphql';
// import { Apollo } from 'apollo-angular';
import { Router } from '@angular/router';

@Injectable({
  providedIn: 'root'
})
export class AuthService {
  // expose all data

  public authErrorMessages$: Observable<GraphQLError[]>;
  public isLoading$: Observable<boolean>;
  public user$: Observable<User>;

  constructor(
    private router: Router,
    // private apollo: Apollo,
    private currentUserGQL: CurrentUserGQL,
    private signupGQL: SignupGQL,
    private loginGQL: LoginGQL,
    private forgotPasswordGQL: ForgotPasswordGQL,
    private resetPasswordGQL: ResetPasswordGQL
  ) {
    const currentUser$ = this.currentUserGQL.watch().valueChanges;

    this.user$ = currentUser$.pipe(map(result => result.data.me));
    this.isLoading$ = currentUser$.pipe(map(result => result.loading));
    this.authErrorMessages$ = currentUser$.pipe(
      map(result => result.errors)
    ) as any; // use any because of a prettier bug
  }

  get authenticated(): boolean {
    return localStorage.getItem('token') !== null;
  }

  public signup({ email, password }) {
    return this.signupGQL.mutate({ email, password });
  }
  public login({ email, password }) {
    return this.loginGQL.mutate({ email, password }).pipe(
      map(result => result.data.login),
      tap(loginData => {
        localStorage.setItem('visual-knight-token', loginData.token.accessToken);
      })
    );
  }

  public forgotPassword(email: string) {
    return this.forgotPasswordGQL.mutate({ email });
  }

  public resetPassword(password: string, token: string) {
    return this.resetPasswordGQL.mutate({ password, token });
  }

  public logout() {
    // this.apollo.getClient().resetStore();
    this.router.navigateByUrl('/user');
  }
}
