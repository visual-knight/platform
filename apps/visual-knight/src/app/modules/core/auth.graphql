query currentUser {
  me {
    ...UserData
  }
}

mutation signup($email: String!, $password: String!) {
  signup(email: $email, password: $password) {
    token {
      ...AuthToken
    }
    user {
      ...UserData
    }
  }
}

mutation login($email: String!, $password: String!) {
  signup(email: $email, password: $password) {
    token {
      ...AuthToken
    }
    user {
      ...UserData
    }
  }
}

fragment AuthToken on AuthToken {
  expiresIn
  accessToken
}

fragment UserData on User {
  id
  active
  email
  apiKey
}
