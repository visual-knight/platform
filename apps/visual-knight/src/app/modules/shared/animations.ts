import {
  trigger,
  animate,
  style,
  transition,
  state,
  animation,
  useAnimation,
  sequence,
  query,
  stagger
} from '@angular/animations';

const reusable = animation(
  [
    style({
      opacity: '{{opacity}}',
      transform: 'scale({{scale}}) translate3d({{x}}, {{y}}, {{z}})'
    }),
    animate('{{duration}} {{delay}} cubic-bezier(0.0, 0.0, 0.2, 1)', style('*'))
  ],
  {
    params: {
      duration: '200ms',
      delay: '0ms',
      opacity: '0',
      scale: '1',
      x: '0',
      y: '0',
      z: '0'
    }
  }
);

export const vkAnimations = [
  trigger('animate', [transition('void => *', [useAnimation(reusable)])]),

  trigger('fadeInOut', [
    state(
      '0',
      style({
        opacity: 0,
        display: 'none'
      })
    ),
    state(
      '1',
      style({
        opacity: 1,
        display: 'block'
      })
    ),
    transition('0 => 1', animate('300ms')),
    transition('1 => 0', animate('300ms'))
  ])
];

export const rowsAnimation = trigger('rowsAnimation', [
  transition('void => *', [
    useAnimation(
      animation(
        [
          style({ opacity: '0' }),
          sequence([animate('{{duration}} {{delay}}', style({ opacity: 1 }))])
        ],
        {
          params: {
            duration: '300ms',
            delay: '0ms'
          }
        }
      )
    )
  ])
]);

export const filterAnimation = trigger('filterAnimation', [
  transition(':enter, * => 0, * => -1', []),
  transition(':increment', [
    query(
      ':enter',
      [
        style({ opacity: 0, width: '0px' }),
        stagger(50, [
          animate('300ms ease-out', style({ opacity: 1, width: '*' }))
        ])
      ],
      { optional: true }
    )
  ]),
  transition(':decrement', [
    query(':leave', [
      stagger(50, [
        animate('300ms ease-out', style({ opacity: 0, width: '0px' }))
      ])
    ])
  ])
]);
