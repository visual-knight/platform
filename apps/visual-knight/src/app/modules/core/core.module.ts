import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { AuthService } from './auth-service.service';
import { TestHelperService } from './services/test-helper.service';
import { GraphQLModule } from './graphql.module';

@NgModule({
  declarations: [],
  providers: [AuthService, TestHelperService],
  imports: [CommonModule, GraphQLModule]
})
export class CoreModule {}
