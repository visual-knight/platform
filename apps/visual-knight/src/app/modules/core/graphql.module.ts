import { NgModule } from '@angular/core';
import { ApolloModule, APOLLO_OPTIONS, Apollo } from 'apollo-angular';
import { HttpLinkModule, HttpLink } from 'apollo-angular-link-http';
import { InMemoryCache } from 'apollo-cache-inmemory';
import { environment } from '../../../environments/environment';
import { ApolloLink } from 'apollo-link';
import { HttpHeaders } from '@angular/common/http';
import { onError } from 'apollo-link-error';
import { Router } from '@angular/router';

const uri = environment.graphql.uri; // <-- add the URL of the GraphQL server here
export function createApollo(httpLink: HttpLink, router: Router) {
  const http = httpLink.create({ uri });

  const authMiddleware = new ApolloLink((operation, forward) => {
    const token = localStorage.getItem('visual-knight-token');
    if (!token) return forward(operation);

    operation.setContext({
      headers: new HttpHeaders().set('Authorization', `Bearer ${token}`)
    });

    return forward(operation);
  });

  const logoutLink = onError(({ graphQLErrors }) => {
    if (
      graphQLErrors[0].extensions.code === 'INTERNAL_SERVER_ERROR' &&
      graphQLErrors[0].extensions.exception.status === 401
    ) {
      localStorage.removeItem('visual-knight-token');
      router.navigateByUrl('/user');
    }
  });

  return {
    link: authMiddleware.concat(logoutLink.concat(http)),
    cache: new InMemoryCache()
  };
}

@NgModule({
  imports: [ApolloModule, HttpLinkModule],
  exports: [ApolloModule, HttpLinkModule],
  providers: [
    {
      provide: APOLLO_OPTIONS,
      useFactory: createApollo,
      deps: [HttpLink, Router]
    }
  ]
})
export class GraphQLModule {}
