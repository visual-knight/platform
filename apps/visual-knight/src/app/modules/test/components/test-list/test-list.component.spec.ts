import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import {
  MatTableModule,
  MatIconModule,
  MatMenuModule
} from '@angular/material';
import { RouterTestingModule } from '@angular/router/testing';
import { ApolloTestingModule } from 'apollo-angular/testing';
import { TestListComponent } from './test-list.component';
import { of } from 'rxjs';
import { TestDataFragment } from '../../../core/types';
import { TestsDataSource } from '../../test.datasource';
import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
import { By } from '@angular/platform-browser';
import { DebugElement } from '@angular/core';
import { TestService } from '../../services/test.service';

describe('TestListComponent', () => {
  let component: TestListComponent;
  let fixture: ComponentFixture<TestListComponent>;
  let router: RouterTestingModule;
  let testService: TestService;

  const test: TestDataFragment = {
    id: '12',
    name: 'name 1',
    variations: [
      {
        id: '33',
        browserName: 'Chrome',
        deviceName: 'Device name',
        createdAt: Date.now(),
        testSessions: []
      }
    ],
    project: {
      id: '44'
    }
  };
  const testList: TestDataFragment[] = [test];

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [TestListComponent],
      imports: [
        MatTableModule,
        MatIconModule,
        RouterTestingModule,
        MatMenuModule,
        ApolloTestingModule,
        BrowserAnimationsModule
      ]
    }).compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(TestListComponent);
    router = TestBed.get(RouterTestingModule);
    testService = TestBed.get(TestService);
    component = fixture.componentInstance;
    component.dataSource = new TestsDataSource(of(testList));
    fixture.detectChanges();
  });

  it('should render not empty test list', () => {
    expect(fixture).toMatchSnapshot();
  });

  it('should delete test', () => {
    const onDeleteTest = jest.spyOn(component, 'onDeleteTest');
    const removeTest = jest.spyOn(testService, 'removeTest');


    const rowElement: DebugElement = fixture.debugElement.query(
      By.css('[id*=test_row]')
    );
    rowElement.query(By.css('#actionMenuBtn')).nativeElement.click();
    rowElement.query(By.css('#deleteBtn')).nativeElement.click();

    expect(onDeleteTest).toHaveBeenCalledWith(test);
    expect(removeTest).toHaveBeenCalledWith(test.id);
  });
});
