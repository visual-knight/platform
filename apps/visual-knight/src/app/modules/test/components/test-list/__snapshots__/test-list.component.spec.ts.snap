// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TestListComponent should render not empty test list 1`] = `
<visual-knight-test-list
  dataSource={[Function TestsDataSource]}
  displayedColumns={[Function Array]}
  testService={[Function TestService]}
>
  <mat-table
    class="mat-table"
    ng-reflect-data-source="[object Object]"
    role="grid"
  >
    
    <mat-header-row
      class="mat-header-row ng-tns-c0-0 ng-star-inserted"
      role="row"
    >
      
      <mat-header-cell
        class="mat-header-cell cdk-column-name mat-column-name ng-tns-c0-0 ng-star-inserted"
        role="columnheader"
      >
        Testname
      </mat-header-cell>
      <mat-header-cell
        class="mat-header-cell cdk-column-variations mat-column-variations ng-tns-c0-0 ng-star-inserted"
        role="columnheader"
      >
        Results
      </mat-header-cell>
      <mat-header-cell
        class="mat-header-cell cdk-column-viewVariations mat-column-viewVariations ng-tns-c0-0 ng-star-inserted"
        role="columnheader"
      />
    </mat-header-row>
    
    <mat-row
      class="mat-row ng-tns-c0-0 ng-trigger ng-trigger-rowsAnimation ng-star-inserted ng-animating"
      id="test_row_12"
      ng-reflect-router-link="/variation,12"
      role="row"
      style="animation: 300ms linear 0ms 1 normal forwards gen_css_kf_0; animation-play-state: running;"
      tabindex="0"
    >
      
      <mat-cell
        class="mat-cell cdk-column-name mat-column-name ng-tns-c0-0 ng-star-inserted"
        role="gridcell"
        style=""
      >
         name 1 
      </mat-cell>
      <mat-cell
        class="mat-cell cdk-column-variations mat-column-variations ng-tns-c0-0 ng-star-inserted"
        role="gridcell"
        style=""
      >
         1 test(s) 
      </mat-cell>
      <mat-cell
        class="mat-cell cdk-column-viewVariations mat-column-viewVariations ng-tns-c0-0 ng-star-inserted"
        role="gridcell"
        style=""
      >
        <div
          class="actions"
          fxlayout="row"
        >
          <button
            aria-haspopup="true"
            class="mat-menu-trigger"
            id="actionMenuBtn"
            mat-icon-button=""
            ng-reflect-menu="[object Object]"
            type="button"
          >
            <mat-icon
              aria-hidden="true"
              class="icon mat-icon notranslate material-icons mat-icon-no-color"
              role="img"
            >
              more_vert
            </mat-icon>
          </button>
        </div>
        <mat-menu
          class="ng-tns-c0-0 ng-tns-c3-1"
        >
          
        </mat-menu>
      </mat-cell>
    </mat-row>
    
  </mat-table>
</visual-knight-test-list>
`;
